### PR TITLE
shrink pending command queue when needed in apply delegates

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
Shrink the pending command queue in apply worker, prevent potential memory usage increasing.

### CODE REVIEW

File: src/raftstore/store/worker/apply.rs [hunk 1]
```diff
@@ -55,6 +55,7 @@ use super::metrics::*;
 
 const WRITE_BATCH_MAX_KEYS: usize = 128;
 const DEFAULT_APPLY_WB_SIZE: usize = 4 * 1024;
+const SHRINK_PENDING_CMD_QUEUE_CAP: usize = 64;
```
Comment (BusyJay) [inline src/raftstore/store/worker/apply.rs]: Is the value suitable?

Comment (zhangjinpeng1987) [reply src/raftstore/store/worker/apply.rs]: This value is adjusted with SHRINK_CACHE_CAPACITY which is 64, i think there is at most one pending command relate to one entry?

**Comment by ngaut** [comment]
Are you fixing memory leak? please add description for your purpose.

**Pull request merged**